lib/spack/spack/ci.py

```diff
@@ -721,6 +721,10 @@ def generate_gitlab_ci_yaml(env, print_summary, output_file,
 
     output_object['stages'] = stage_names
 
+    # TODO(opadron): remove this or refactor
+    import spack.ci_optimization.optimizer as optimizer
+    output_object = optimizer(output_object)
```

Comment:
Can this be turned into an optional pass that defaults to `False`?  If so, I can definitely get behind this.  It would make it so spack has a way around the gitlab yaml size limit, but when they fix that, removal of this option would be easy.